### PR TITLE
Initial work toward new component bundle directories

### PR DESCRIPTION
**Pull request by lmorchard**
Some remaining TODOs:
- [x] Tests for new IncompatibleAddons component
- [x] Fix / move a broken ExperimentPage test based on IncompatibleAddons render
- [x] Tweak eslint rules for test modules using devDependencies
- [x] #2808 <strike>Actually migrate over some styles for the new bundles components</strike>
- [x] Clean up some commented out bits
- [x] Fix flow type validation failures
- [x] Docs based on https://gist.github.com/chuckharmston/01e5b7a16ba9b771863cbb8b2de64138

Done so far:

- Moved ExperimentRowCard and UpdateList components into subdirectories

- Extracted IncompatibleAddons as a component from ExperimentPage

- Webpack config to extract static/app/app.js.css from component imports

- Link to static/app/app.js.css styles from the static page template

- Storybook config changes to use frontend/src/app/**/{ComponentName}/stories.jsx files

- Mocha test command changes use frontend/src/app/**/{ComponentName}/tests.js

- Hacks to frontend/tasks/pages.js to ignore scss imports

- Hacks to test-setup.js to ignore scss imports

See also:
https://gist.github.com/chuckharmston/01e5b7a16ba9b771863cbb8b2de64138

Issue #2807

### CODE REVIEW

**Comment by lmorchard** [comment]
@fzzzy - Still got some work to do here, but I think this is doing the right things so far with styles, stories, & tests.

**Comment by fzzzy** [comment]
What on earth is going on in this failure here? o.O

File: frontend/src/app/components/IncompatibleAddons/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const path = require('path');
+
+module.exports = {
+  module: {
+    rules: [
+      {
+        test: /\.(css|scss)$/,
+        loaders: ["style-loader", "css-loader", "sass-loader"],
```
Comment (fzzzy) [inline frontend/src/app/components/IncompatibleAddons/webpack.config.js]: The sass-loader docs gave something very different than this. I never would have figured this out...

Comment (lmorchard) [reply frontend/src/app/components/IncompatibleAddons/webpack.config.js]: Ignore that... it didn't work and shouldn't have been there - I was editing the wrong file in the wrong dir & accidentally committed it

**Comment by lmorchard** [comment]
> What on earth is going on in this failure here? o.O

Looks like the test modules are getting caught up in the <strike>webpack bundling</strike> JS linting with browser rules. But, I've seen that before and can fix

File: frontend/src/app/components/IncompatibleAddons/webpack.config.js (hunk 1 above)
Comment (lmorchard) [reply frontend/src/app/components/IncompatibleAddons/webpack.config.js]: FWIW, these are the changes to the main webpack config, basically from the sass-loader docs: https://github.com/mozilla/testpilot/pull/2859/files#diff-11e9f7f953edc64ba14b0cc350ae7b9d

**Comment by fzzzy** [comment]
Looking at the diff, I'm guessing the hacks you have in test-setup were what I needed to fix the weird problem I had trying to load sass earlier today.

I like the look of the way webpack is set up here better than what I had, too.

File: frontend/src/app/components/IncompatibleAddons/webpack.config.js (hunk 1 above)
Comment (fzzzy) [reply frontend/src/app/components/IncompatibleAddons/webpack.config.js]: Yeah that changeset looks really good. I think I can actually understand how it works!

**Comment by lmorchard** [comment]
Okay. Reorg'd this into a few more self-contained commits and cleaned some things up. And yay the tests pass!

<strike>Thinking I might punt on actually migrating styles over for now, though: It's going to take a bit more SCSS hacking than just moving the style files</strike>

Looks like we have #2808 for porting over SCSS - I'll say this PR *enables* index.scss per component directory, but we still have to rework the existing SCSS a bit to *use* the capability.

**Comment by fzzzy** [comment]
I think this is ready to land. I would be fine with just linking to the gist in the documentation for now.

**Review by fzzzy — APPROVED** [review]
This is going to be very nice.

**Comment by lmorchard** [comment]
This could use some revision, [but I just copypasta'd most of that proposal into the Storybook docs](https://github.com/lmorchard/testpilot/blob/e443d7fe4aecad48153705bc157707033d4d803e/docs/development/storybook.md). Might just call that good enough and merge after tests go green

**Comment by fzzzy** [comment]
Seems good enough for now!

**Pull request merged**